### ISSUE TITLE
Ridge Racer 7 no longer working, but previous build was perfectly playable

### ISSUE THREAD

**Issue by Edenia3** [RPCS3/rpcs3#9555]
Just nothing happens after intro, game hangs with v0.0.14-11518-6ae1f1b0 Alpha
Copied back previous build and it's working perfectly fine.

**Comment by elad335** [comment]
I can't do anything with this report without log, this is the bare minimum.

**Comment by Edenia3** [comment]
> 
> 
> I can't do anything with this report without logs from before and after the regression, this is the bare minimum.

Ok, i found the issue - it's the Latest Nvidia Beta drivers 460.97
Rolled back to 460.89 and now it's fine

**Comment by legend800** [comment]
It's not a driver issue. Still happens with 460.89. Something with the emu. Here's a log. Let me know if you need anything more to help debug. 

[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/5778976/RPCS3.zip)

**Comment by Edenia3** [comment]
> 
> 
> It's not a driver issue. Still hangs on menus with 460.89. Something with the emu. It's random and happens 5/10 times. Here's a log. Let me know if you need anything more to help debug.
> 
> [RPCS3.zip](https://github.com/RPCS3/rpcs3/files/5778976/RPCS3.zip)

No, my RR7 works like a charm, my issue was a bad Vulkan driver in 460.97. I'm playing with Ridge Racer 7 PLATINUM EDITION BCES00009

**Comment by elad335** [comment]
Try libvdec.sprx LLE.

**Comment by Augusto7743** [comment]
RR7 Demo USA works fine

**Comment by Back2Life888** [comment]
RR7 really doesn't work.. Back then, it works like a charm but with eye sore bugs like the famous Black screen Sunset Heights, black screen Unlock machine scene and flickering trees. But now, with those bugs fixed, It crashes every time. I was using the 3D NPUB version. I played the game using Vulkan, but because of the crashes, I never get past RSGP 13. So I switched to OpenGL. It worked but once I was done in RSGP 13, it crashes once again.

My specs:
Intel i7-6700 3.40GHz (8 CPUs) (Not overclocked)
NViDiA Palit 1650 Dual 4 GB (Not overclocked)
32 GB DDR4 RAM

Does anyone know another fix other than rolling back GPU drivers?? Because rolling back GPU drivers makes my other games not work at all..

**Comment by legend800** [comment]
I still believe that this game **_has_** always had this random freeze issue in the menus (for years) and it takes a few restarts to get past it sometimes. Not a gfx driver thing as I doubt that this behavior would be exhibited on a single game in a sporadic fashion like this. 

I'm on the latest drivers and booted the game 3 times before getting past it. 

![2021-01-16 09_43_32-FPS_ 59 79 _ Vulkan _ 0 0 14-11506 _ Ridge Racer7 3D License Ver (Full Game)  NP](https://user-images.githubusercontent.com/20983844/104819139-63960900-57e0-11eb-98bd-88f88be659fe.png)

**Comment by Asinin3** [comment]
Yeah I had these crashes too a year ago, I don't think it's a regression

**Comment by Back2Life888** [comment]
> Try libvdec.sprx LLE.

I had it enabled and it give the game an extra "oomph" for the first run.. After one race, the extra "oomph" ran out and eventually, it crashed..

**Comment by Back2Life888** [comment]
I have an idea.. We find out what is the first build that the crashes took place.. Then we examine the codes and lines that was changed..

**Comment by AniLeo** [comment]
If this is a regression, need to know the exact build that introduced it

**Comment by Back2Life888** [comment]
> If this is a regression, need to know the exact build that introduced it

If we need to find a build, you also need to help.. We are about to search more than 100 builds so we need help..

**Comment by Asinin3** [comment]
Testing is up to users who report the issues. Besides, it's not like you have to test **every build** use some common sense and pick a build from e.g 6 months ago. If it's still broken go to a build from 12 months ago... by using the process of elimination you can find problems quickly. Also remember to use same RPCS3 folder so you can use same cache's and config files.

**Comment by Back2Life888** [comment]
> Testing is up to users who report the issues. Besides, it's not like you have to test **every build** use some common sense and pick a build from e.g 6 months ago. If it's still broken go to a build from 12 months ago... by using the process of elimination you can find problems quickly. Also remember to use same RPCS3 folder so you can use same cache's and config files.

Thats exactly what im doing.. Im now about to search only 2 Months of builds because I did the thing you just said..

**Comment by Back2Life888** [comment]
I got the origins of the build where the game freezes.. Its 0.0.10-10432 that introduced the freezing.. But the crashes are nowhere to be seen in this build so I'm going to research again..

**Comment by Back2Life888** [comment]
Alright, I think I give up.. The bugs switches on and off on builds.. It's basically a point that the elimination process is useless.. All builds have different results.. Sometimes the old build crashes but a new build doesn't crash and It will basically make it hard to find the main build that caused it..

**Comment by legend800** [comment]
Exactly, glad you finally came to the conclusion I did years ago and posted above :)

**Comment by Back2Life888** [comment]
Alright once again as I found out that the crashes happen also at old builds.. It also crashes even before the trees or Sunset Heights are fixed.. It also even crashes on builds before the Sunset Heights workaround has been found.. The newer versions are just more severe with crashing.. The old builds does crash but just less and isn't much of a problem at all.. This is not a regression anymore.. This is an old bug...

**Comment by postalesdeamor** [comment]
In my case works intermittently, like #11134

This did not happen in previous versions of RPCS3 (although I do not know in which version it started to crash).

E SELF: SELF: Failed to decrypt metadata info! E SELF: SELF: Failed to load SELF file metadata! E SYS: Failed to decrypt SELF: D:/hyperspin/Emulators/RPCS3/dev_hdd0/game/BCES00009/USRDIR/main.self F sys_process: Failed to boot from exitspawn! (path="D:/hyperspin/Emulators/RPCS3/dev_hdd0/game/BCES00009/USRDIR/main.self", error=Failed to decrypt content)

**Comment by mrwallace888** [comment]
I went to play like back in early January or late December and RR7 worked perfectly fine. Hadn't played in a while, decided to hop on, updated RPCS3, and while RR7 still works, Sunset Heights now has a black screen, whereas it worked before.

**Comment by legend800** [comment]
So much noise in this issue. This is just about your shader caching selection. Reboot and it's probably fine else try the other settings. 

There's been no change to emulation on this game in years, other than to fix the Sunset heights colors. Works perfectly fine for me and has for a very long time.

**Comment by postalesdeamor** [comment]
@legend800 in my case is not a  shader caching problem. The USA edition works perfect but EU not, it works intermittently, firs load NOT, second YES, etc.

**Comment by Megamouse** [comment]
needs a retest.

**Comment by Ordinary205** [comment]
Ridge Racer 7 seems to be running fine, even if I booted this game for the first time I still haven't encountered any strange issues.
![Ridge Racer 7 1](https://github.com/RPCS3/rpcs3/assets/125112932/c3f6164a-50ca-4658-859b-410a6399c0fc)
![Ridge Racer 7 2](https://github.com/RPCS3/rpcs3/assets/125112932/3b684189-646f-4777-854f-6ec07181a115)
![Ridge Racer 7 3](https://github.com/RPCS3/rpcs3/assets/125112932/76c0ed14-3ae8-4f2d-9ae5-5ad0659e5617)
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/13583635/RPCS3.log.gz)

**Comment by Back2Life888** [comment]
Good CPUs actually remove the in game crashes but it doesn't remove the crashes when you scroll too fast on the menu. CPUs that are weaker crash in game though.

**Comment by digant73** [comment]
is there any survivor in this thread that can provide an update on this issue?

**Comment by antoniodesousa** [comment]
I played the game for around 30 minutes and I couldn't reproduce any of the issues described in this ticket.
Although, it seems to log quite a LOT of stuff to the log file (757 MB uncompressed).

[RPCS3.log.gz](https://drive.google.com/file/d/1wBAOQLk1uu9M3RzUE7KFCN2HPRe0KXnX/view?usp=drive_link)

**Comment by digant73** [comment]
it seems this issue has been better covered and also fixed in #13991 (reported in the wiki). PPU Interpreter seems the option to enable to fix this issue. I see there is RR7 Demo, so I will probably test it this night and report here the results.

**Comment by digant73** [comment]
I tested the demo, replicated the issue on old builds, tested on current build 0.0.38-18263, verified that PPU Interpreter (also reported in wiki page) is required to avoid a possible soft freeze on main menu, As also reported by other users some days ago, the issue is likely no more present and can be closed by main developers
